### DISCUSSION TITLE
Configured volumeGrowthCount, but it did not take effect.

### DISCUSSION THREAD

**Discussion by photosssa** [seaweedfs/seaweedfs#6312] [Q&A] [answered]
I have a configuration of 2 SSDs and 2 HDDs, and I have directed all writes to the SSD using fs.configure. I want to migrate the volumes on the SSD to the HDD in a shorter total write time, so I set volumeGrowthCount=1. According to my understanding, no additional volumes should be created on the SSD before a single volume reaches the 30GB limit. I created a test program that writes 10,000 files of 1MB each using 16 threads, and initially, there was indeed only one volume, which I then moved to the HDD. However, after repeating the test several times, I consistently observed that three additional volumes were created at certain moments, and subsequently, file writes would be evenly distributed across the three volumes, which does not align with my expectations. Why this is happening?
```
> fs.configure
{
  "version":  0,
  "locations":  [
    {
      "locationPrefix":  "/",
      "collection":  "",
      "replication":  "001",
      "ttl":  "",
      "diskType":  "ssd",
      "fsync":  false,
      "volumeGrowthCount":  1,
      "readOnly":  false,
      "dataCenter":  "",
      "rack":  "",
      "dataNode":  "172.23.156.57:8080",
      "maxFileNameLength":  0,
      "disableChunkDeletion":  false,
      "worm":  false
    }
  ]
}
```
```
volume.tier.move -fromDiskType=ssd -toDiskType=hdd -fullPercent=2 -quietFor=0s -toReplication=001 -force
```
```
Disk ssd(volume:7/8 active:7 free:1 remote:0)
          volume id:6  size:8  version:3  compact_revision:1  modified_at_second:1732871056  disk_type:"ssd"
          volume id:8  size:8  version:3  compact_revision:1  modified_at_second:1732871066  disk_type:"ssd"
          volume id:7  size:8  version:3  compact_revision:1  modified_at_second:1732871076  disk_type:"ssd"
          volume id:9  size:8  version:3  compact_revision:1  modified_at_second:1732871086  disk_type:"ssd"
          volume id:15  size:1089520344  file_count:1039  replica_placement:1  version:3  modified_at_second:1732874282  disk_type:"ssd"
          volume id:14  size:1118881816  file_count:1067  replica_placement:1  version:3  modified_at_second:1732874282  disk_type:"ssd"
          volume id:16  size:1162924024  file_count:1109  replica_placement:1  version:3  modified_at_second:1732874282  disk_type:"ssd"
        Disk ssd total size:3371326216 file_count:3215
```

**Comment by kmlebedev** [comment]
Additional volumes seem to be created for logs, in any case, you can enable debug logging and see where the volume creation is initialized

Also, please highlight what problem you are trying to solve

**Reply by photosssa** [reply]
If the creation of additional volumes due to logging is acceptable, it's fine as long as my test program only writes to one volume. The current situation is that my test program's writes are no longer limited to a single volume.

**Comment by photosssa** [comment]
I want to use the SSD as a read-write cache, so I hope to quickly move newly written data from the SSD to the HDD. If my strategy is to move data once every 10GB of new writes on the SSD, the default volume growth count of 7 would result in a total volume count in the system being seven times larger compared to 1. Additionally, I will quickly convert the volumes on the HDD to EC (Erasure Coding), which will allow the read concurrency of EC shards to replace the concurrency of the volumes. Therefore, I believe that in my plan, a volume growth count of 1 would be more optimal. However, it seems that I am unable to achieve this stable behavior.

**Reply by photosssa** [reply, marked as answer]
I added the collection parameter for writes using fs.configure, and now it seems to exhibit the behavior I wanted: it will always create only one new volume, and regardless of how many writable volumes there are, all writes will be concentrated in this volume. I realized that I previously misunderstood the meaning of the volume growth count parameter; it indicates the number of new volumes to create when there are no suitable writable volumes. The write operations are always evenly distributed across all suitable writable volumes and are not affected by the growth count.
```
fs.configure -locationPrefix=/ -collection=benchmark -apply
```
